backend/src/app/binfile/v2.clj

```diff
@@ -0,0 +1,714 @@
+                           (let [features (cfeat/check-supported-features! features)]
+                             (-> (::features cfg)
+                                 (set/difference cfeat/frontend-only-features)
+                                 (set/union features))))))
+
+        _    (when (contains? cf/flags :file-schema-validation)
+               (fval/validate-file-schema! file))
+
+        _    (when (contains? cf/flags :soft-file-schema-validation)
+               (let [result (ex/try! (fval/validate-file-schema! file))]
+                 (when (ex/exception? result)
+                   (l/error :hint "file schema validation error" :cause result))))
+
+        file (if (contains? (:features file) "fdata/objects-map")
+               (feat.fdata/enable-objects-map file)
+               file)
+
+        file (if (contains? (:features file) "fdata/pointer-map")
+               (binding [pmap/*tracked* (pmap/create-tracked)]
+                 (let [file (feat.fdata/enable-pointer-map file)]
+                   (feat.fdata/persist-pointers! cfg (:id file))
+                   file))
+               file)]
+
+    (db/insert! conn :file
+                (-> file
+                    (update :features db/encode-pgarray conn "text")
+                    (update :data blob/encode))
+                {::db/return-keys false}))
+
+  (doseq [thumbnail (read-seq cfg :file-object-thumbnail file-id)]
+    (let [thumbnail (-> thumbnail
+                        (update :file-id lookup-index)
+                        (update :media-id lookup-index))
+          file-id    (:file-id thumbnail)
+
+          thumbnail (update thumbnail :object-id
+                            #(str/replace-first % #"^(.*?)/" (str file-id "/")))]
+
+      (db/insert! conn :file-tagged-object-thumbnail thumbnail
+                  {::db/return-keys false})))
+
+  (doseq [rel (read-obj cfg :file-rels file-id)]
+    (let [rel (-> rel
+                  (update :file-id lookup-index)
+                  (update :library-file-id lookup-index)
+                  (assoc :synced-at timestamp))]
+      (db/insert! conn :file-library-rel rel
+                  ::db/return-keys false)))
+
+  (doseq [media (read-seq cfg :file-media-object file-id)]
+    (let [media (-> media
+                    (update :id lookup-index)
+                    (update :file-id lookup-index)
+                    (update :media-id lookup-index)
+                    (update :thumbnail-id lookup-index))]
+      (db/insert! conn :file-media-object media
+                  ::db/return-keys false))))
+
+(def ^:private empty-summary
+  {:teams #{}
+   :files #{}
+   :projects #{}
+   :file-media-objects #{}
+   :team-font-variants #{}
+   :storage-objects #{}})
+
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;; PUBLIC API
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+
+(defn export-team!
+  [cfg team-id]
+  (let [id  (uuid/next)
+        tp  (dt/tpoint)
```

Comment:
I think excesive abbreviations do harm readability of the code. Some very common ones like `cfg` for config may be valid. But this one for example causes me to stop reading to check what is a `tpoint`.

I think it's very easy to write `time-point` or `log/info` and makes the code significantly easier to read. Taking into account that any code is read much more times than it's written, it pays the effort... :stuck_out_tongue:

Reply:
Agree

But concretly `tp` is used many times in the application, it is not new and not new to this ns.

Reply:
Haha... of course not from this one. it was only a general rant :D